src/Adapter/Product/Repository/ProductRepository.php

```diff
@@ -285,25 +285,41 @@ public function get(ProductId $productId): Product
         return $this->loadProduct($product);
     }
 
+    /**
+     * @param ProductId $productId
+     *
+     * @return ProductType
+     *
+     * @throws CoreException
+     * @throws Exception
+     * @throws ExceptionAlias
+     * @throws ProductConstraintException
+     * @throws ProductNotFoundException
+     */
     public function getProductType(ProductId $productId): ProductType
     {
-        $qb = $this->connection->createQueryBuilder();
-        $qb
-            ->addSelect('p.product_type')
+        $result = $this->connection->createQueryBuilder()
+            ->addSelect('p.product_type, p.cache_is_pack, p.is_virtual, p.cache_default_attribute')
```

Comment:
Colums `p.cache_is_pack, p.is_virtual, p.cache_default_attribute` are added but look not used. Can you please explain why you added them ?

Reply:
yes, indeed, thats a leftover. (I've added them previously, because I tried another approach of fetching the values needed to determine the product type without loading the whole product). But now they are useless, so I'll remove them.